### ISSUE TITLE
Service Task hangs infinitely in the active state if it is after a subprocess.

### ISSUE THREAD

**Issue by CEMEHEX** [kiegroup/kogito-runtimes#2467]
### Describe the bug

If a Service Task is in the bpmn schema after Reusable, it will hang in the active state indefinitely when the process is started, even though the Java method of the class that is called by the Service Task is working successfully in this case.
<img width="1332" alt="image" src="https://user-images.githubusercontent.com/11317249/190195760-cde5a5d8-6e38-4507-a600-d3ded7f9d27c.png">

### Expected behavior

Service Task must successfully complete its execution and move on to the next step, the presence of a subprocess before it should not affect its behavior.

### Actual behavior

Service Task infinitely hangs in active state if there is a subprocess before it, and successfully works if there is no subprocess.

### How to Reproduce?

1. Create a bpmn diagram as on the screenshot in the description.
2. Create Java class with empty method and bind it to Service Task.
3. Create a process that does nothing, bind it to Reusable.
4. User Task can be left empty.
5. Build, run the process, see its status in Management Console.

### Output of `java -version`

17.0.3

### Kogito version or git rev (or at least Quarkus version if you are using Kogito via Quarkus platform BOM)

1.23.0.Final; 1.25.0.Final; 1.27.0.Final

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.6.3

**Comment by tiagodolphine** [comment]
Opened the https://issues.redhat.com/browse/KOGITO-7945 to track the issue.
@ricardozanini fyi.

**Comment by CEMEHEX** [comment]
Looks like this issue was solved in 1.29.0.Final

**Comment by ricardozanini** [comment]
@tiagodolphine can you confirm @CEMEHEX statement? 

@CEMEHEX, you tested on your end, and it's working now? 🤔

**Comment by CoffeeWithBurger** [comment]
@ricardozanini @tiagodolphine 
The problem is still reproducing. An important factor for its reproduction is the connection of the **kogito-addons-springboot-persistence-postgresql** dependency

Described in more detail in the chat https://kie.zulipchat.com/#narrow/stream/232676-kogito/topic/Frieze.20of .20the.20process

**Comment by tiagodolphine** [comment]
good point, it is probably something related to the persistence addon, are there any exceptions in the log?

**Comment by CoffeeWithBurger** [comment]
> good point, it is probably something related to the persistence addon, are there any exceptions in the log?

There are no exceptions anywhere. The process silently stops in place.

**Comment by CoffeeWithBurger** [comment]
@tiagodolphine 
Are there any recommendations on how to fix this problem?
Or how long to wait for the bug fix?
So far, the idea is only to replace postgres with infinispan and test its functionality with its plugin. But I would not like to resort to this, since this is another pod. It is more convenient to keep all projects in one postgres than to raise different databases for different services.

**Comment by tiagodolphine** [comment]
@CoffeeWithBurger we have this on the backlog to take a look, maybe in the next week, for now I think you can try the `kogito-addons-springboot-persistence-jdbc` that supports PostgreSQL as well, and check it the issue does not happen with it.

**Comment by CoffeeWithBurger** [comment]
@tiagodolphine
Changed dependencies to spring-boot-starter-jdbc + postgresql + kogito-addons-springboot-persistence-jdbc. The result is the same. The process still stops silently.

**Comment by CoffeeWithBurger** [comment]
@tiagodolphine 
In the management-console, I observe that the service-task remains in the active state, although the handler has already completely completed its work.

![image](https://user-images.githubusercontent.com/105427130/223070313-a02b2ff3-05b5-4300-9d65-1cbfdeff50ae.png)
![image](https://user-images.githubusercontent.com/105427130/223070527-378005f2-58ce-4bc0-8067-3b5cecccba02.png)

It doesn't matter which handler is in the service-task, it won't be able to terminate and the process will stop.

**Comment by CoffeeWithBurger** [comment]
@ricardozanini @tiagodolphine 
UPD: this problem is reproduced on all variants with persistence. For example, with MongoDB (spring-boot-starter-data-mongodb + kogito-addons-springboot-persistence-mongodb), this problem is also reproduced.
I don't see any possible solutions to this problem. @tiagodolphine maybe you have ideas on how to fix this problem, except disable persistence.

**Comment by tiagodolphine** [comment]
@CoffeeWithBurger thanks for the further analysis, I'll update the issue with this information in the https://issues.redhat.com/browse/KOGITO-7945

**Comment by CoffeeWithBurger** [comment]
I checked this behavior on quarkus. The result is the same. It is enough to add a service-task call to the test schemes after calling the subprocess.
Used this example: https://github.com/kiegroup/kogito-examples/tree/stable/kogito-quarkus-examples/process-postgresql-persistence-quarkus
Changed the scheme as follows: 
![image](https://user-images.githubusercontent.com/105427130/224039115-9b7884b9-5b11-425d-a757-5ac6b24c770b.png)
Added a log at the beginning and a service-task after the subprocess.
![image](https://user-images.githubusercontent.com/105427130/224039240-4b3a352c-366f-4fca-85f6-8c68a9e0a566.png)
The "Print review of the deal" block will no longer be called.
Tested on version 1.35.0.Final. But on previous versions, the case is reproduced in the same way.
